src/Sylius/Component/Promotion/Provider/PreQualifiedPromotionsProviderInterface.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Component\Promotion\Provider;
+
+use Sylius\Component\Promotion\Model\PromotionInterface;
+use Sylius\Component\Promotion\Model\PromotionSubjectInterface;
+
+/**
+ * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
+ */
+interface PreQualifiedPromotionsProviderInterface
```

Comment:
In fact, it's not implementation detail. ``Active`` or ``ActiveByChannel`` is. Interface describes the behaviour of service, it provides pre-qualified promotions to use them somewhere else. Sure, you can implement this interface with ``PromotionsProvider`` and return all promotions, but in this case:
- you need all promotions (they **are** pre-qualified, but all fits your needs), or
- you use a bad service in your implementation, or
- your service implements wrong interface

Having the same method does not determine that these interface are equal. If it could be ``PromotionsProviderInterface`` why not just ``ProviderInterface`` with ``provide()`` method? ;) It provides things.

In my opinion ``PreQualifiedPromotionsProviderInterface`` perfectly says about behaviour of service, without telling anything about implementation (how these promotions are provided).